astropy/wcs/wcs.py

```diff
@@ -129,6 +130,8 @@
 # Additional relax bit flags
 WCSHDO_SIP = 0x10000
 
+# Regular expression defining SIP keyword
+SIP_KW = re.compile('''^(A|B|AP|BP)_(?:[0-9]|1[0-9])_(?:[0-9]|1[0-9])(|[A-Z])$''')
```

Comment:
@Xarthisius Can you clarify what does not match the original SIP paper? Did you mean the optional trailing character? Or the fact that `A_ORDER`, ..., keywords are not removed? Or something else?